File: rogue-server/check/checks/auto_clicker_a.py
from abc import ABC
from util import math_util

from check.auto_clicker_check import AutoClickerCheck


class AutoClickerA(AutoClickerCheck, ABC):

    def __init__(self, data):
        super().__init__(data)

    def handle_check(self):
        if math_util.get_stdev(self.clicks) < .45:
            self.fail()

    def get_size(self):
        return 100
    
    def get_check_name(self):
        return 'AutoClicker'

    def get_check_type(self):
        return 'A'

File: rogue-server/src/main/java/org/hostile/anticheat/check/impl/autoclicker/AutoClickerA.java
package org.hostile.anticheat.check.impl.autoclicker;

import org.hostile.anticheat.check.annotation.CheckMetadata;
import org.hostile.anticheat.check.type.impl.AutoClickerCheck;
import org.hostile.anticheat.data.PlayerData;
import org.hostile.anticheat.util.MathUtil;

@CheckMetadata(type = "AutoClicker", name = "A")
public class AutoClickerA extends AutoClickerCheck {

    public AutoClickerA(PlayerData playerData) {
        super(playerData);
    }

    @Override
    public void handle() {
        debug("stDev", MathUtil.getStandardDeviation(clicks));
    }

    @Override
    public int getSampleSize() {
        return 100;
    }
}
